File: Languages/java/src/provider/MaxValueAnalysis.java
package provider;

import model.SimpleTableAnalysis;
import model.UserInfo;

import java.util.List;

/**
 * Classe de analise que implementa a interface "SimpleTableAnalysis"
 */
public class MaxValueAnalysis implements SimpleTableAnalysis {

    /**
     * Método de interface, responsável por obter o maior valor de credit em uma lista de usuarios
     * @param userInfoList Lista de usuarios
     * @return maior valor de credit
     */
    @Override
    public double analysis(List<UserInfo> userInfoList) {

        double max = Double.MIN_VALUE;
        for(UserInfo userInfo : userInfoList){
            double credit = userInfo.getCredit();
            if(max < credit) max = credit;
        }
        return max;
    }
}

File: Languages/python3/src/provider/MaxValueAnalysis.py
from src.model.SimpleTableAnalysis import SimpleTableAnalysis
from src.model.UserInfo import UserInfo

# Classe de analise que implementa a interface "SimpleTableAnalysis"
class MaxValueAnalysis(SimpleTableAnalysis):

    # Método responsável por obter o maior valor de credit na lista de usuarios
    # @list Lista de usuarios
    # @return Valor máximo da lista
    def analysis(self, userInfoList):
        vmax = -1000000000
        
        for userInfo in userInfoList:
            if(userInfo.getCredit() > vmax): 
                vmax = userInfo.getCredit()

        return vmax